### ISSUE TITLE
Allow changing destination directory name of Cloned Gist Repositories

### ISSUE THREAD

**Issue by JustinGrote** [lostintangent/gistpad#206]
**Is your feature request related to a problem? Please describe.**
Clone to Repository saves as the gist GUID which can make it hard to identify

**Describe the solution you'd like**
When choosing "Clone Repository", offer a popup to name the repository folder in the target rather than the GUID it is typically assigned. This behavior could be toggled by a preference variable

**Describe alternatives you've considered**
Clone and then rename afterwards, however this is less desirable to the workflow

**Comment by lostintangent** [comment]
Ah yeah, that's a great suggestion! In your case, do your gists have descriptions, that you'd want to use as the cloned folder name? At a minimum, we should definitely use the description if the gist has one, since the ID is pretty useless 😁

**Comment by JustinGrote** [comment]
I normally give them descriptions, but it should still prompt with the description auto-populated, because my descriptions tend to be very verbose which I'm not sure if that's the norm or not:
![image](https://user-images.githubusercontent.com/15258962/104988345-af0d0a80-59cc-11eb-910c-bbaef910ba9c.png)

**Comment by lostintangent** [comment]
Got it. Thanks for sharing that detail!

**Comment by JustinGrote** [comment]
@lostintangent I noticed there hasn't been a release in a while but I'm a lot better at vscode extensions (I maintain the PowerShell extension) than I was when I filed this issue, and I'm willing to write a PR if you'd be willing to facilitate it.

I see it being a extension setting you check on, and it'll prompt you for your Repo name after you select the path, auto-populated with your description (sanitized for filesystems and spaces replaced with hyphens)

**Comment by lostintangent** [comment]
Yeah if you're down for creating a PR, that would be much appreciated.

I've been recently focusing my attention on https://gistpad.dev and https://npmjs.com/gistpad-mcp, and so any contributions on the VS Code extension are a huge help 🙏
